test/logging/LoggerFactoryTest.cpp

```diff
@@ -23,18 +23,18 @@ BOOST_AUTO_TEST_CASE(test) {
     Logger& unknown1 = LoggerFactory::getLogger("unknown1");
     Logger& unknown2 = LoggerFactory::getLogger("unknown2");
     Logger& unknown3 = LoggerFactory::getLogger<LoggerFactory>();
-    BOOST_TEST(stdcxx::areSame(typeid(NoopLogger), typeid(unknown1)));
-    BOOST_TEST(stdcxx::areSame(typeid(NoopLogger), typeid(unknown2)));
-    BOOST_TEST(stdcxx::areSame(typeid(NoopLogger), typeid(unknown3)));
-    BOOST_TEST(stdcxx::areSame(unknown1, unknown2));
-    BOOST_TEST(stdcxx::areSame(unknown1, unknown3));
+    BOOST_CHECK(typeid(NoopLogger) == typeid(unknown1));
+    BOOST_CHECK(typeid(NoopLogger) == typeid(unknown2));
+    BOOST_CHECK(typeid(NoopLogger) == typeid(unknown3));
+    BOOST_CHECK(typeid(unknown1) == typeid(unknown2));
+    BOOST_CHECK(typeid(unknown1) == typeid(unknown3));
 
     LoggerFactory::getInstance().addLogger("console", stdcxx::make_unique<ConsoleLogger>());
     Logger& consoleLogger = LoggerFactory::getLogger("console");
     BOOST_TEST(stdcxx::areSame(typeid(ConsoleLogger), typeid(consoleLogger)));
     LoggerFactory::getInstance().removeLogger("console");
     Logger& consoleLogger2 = LoggerFactory::getLogger("console");
-    BOOST_TEST(stdcxx::areSame(typeid(NoopLogger), typeid(consoleLogger2)));
+    BOOST_CHECK(typeid(NoopLogger) == typeid(consoleLogger2));
 
```

Comment:
I disagree: BOOST_CHECK is ok, but with two parameters then.
```
BOOST_CHECK(expected, actual);
```

Reply:
Compilation fails by writing `BOOST_CHECK(typeid(NoopLogger), typeid(consoleLogger2))`:
```
E:\Projet_SRO\rte\windows-porting-final\test\logging\LoggerFactoryTest.cpp(37): warning C4002: trop d'arguments pour l'appel de macro de type fonction 'BOOST_CHECK'
C:\boost-1.72.0\include\boost/test/tools/assertion_result.hpp(57): error C2678: '!' binaireÿ: aucun op‚rateur trouv‚ qui accepte un op‚rande de partie gauche de type 'const BoolConvertable' (ou il n'existe pas de conversion acceptable)
        with
        [
            BoolConvertable=type_info
        ]
```